R/test_description.R

```diff
@@ -83,26 +110,54 @@ as_test_desc <- function(x, type = "call") {
   structure(x, type = type, class = "test_description")
 }
 
+
+
 #' Adds "testthat" style
 #'
 #' @rdname as_test_desc
 as_testthat_desc <- function(x) {
+  if (!is.character(x)) {
+    try_result <- try(eval(x, envir = baseenv()), silent = TRUE)
+    if (!inherits(try_result, "try-error")) x <- try_result
+    else return(as_test_desc(expr_str(x)))
+  }
+
   as_test_desc(x, type = "testthat")
 }
```

Comment:
slight refactor

This little try-catch handler used to live in `test_description_test_that`, but would be reused across all the different calls. This helps to clean up the individual handlers.